### ISSUE TITLE
Implement 'mount' parameter in remaining methods

### ISSUE THREAD

**Issue by Richie765** [kyndryl-open-source/hashi-vault-js#9]
The `mount` parameter is not yet implemented in all Auth Methods and Secret Providers.

**Comment by rod-anami** [comment]
Hi @Richie765, I need to check if all of them can be mounted on different path. I think they can, but need to check that to make sure. Since this is an extra parameter, this is not a breaking change. I'm marking this one for `0.4.1`.

**Comment by Richie765** [comment]
@rod4n4m1, perfect. Shall I make a PR for that? I can also make a few extra tests for it.

**Comment by rod-anami** [comment]
Hi @Richie765, yes, go ahead. I just created the `0.4.1` branch for you. Also, don't forget to add preset values on `Config.js` so, even you don't pass `mount` it will resolve gracefully.

**Comment by rod-anami** [comment]
This is the standard I've used for few auth methods and secret engines.
Config.js
```
kvRootPath: 'secret',
...
pkiRootPath: 'pki',
```

Vault.js
```
let rootPath= "";
    if (mount) {
      rootPath = mount;
    } else if (this.rootPath) {
      rootPath = this.rootPath;
    } else {
      rootPath = config.pkiRootPath;
    }
```

**Comment by rod-anami** [comment]
Hey @Richie765 , no worries, I'll get this done since I did for many functions already.

**Comment by Richie765** [comment]
@rod4n4m1 Thanks a lot, that is awesome. 👍  I'm currently busy with a lot of other things.

**Comment by rod-anami** [comment]
Resolved on `0.4.1 `. Thanks
